### PR TITLE
add ddr regs from sifive

### PR DESCRIPTION
**Pull request by rminnich**
won't compile, I'm doing something dumb, this is a snapshot

### CODE REVIEW

File: src/soc/sifive/fu540/src/ddrregs.rs [hunk 1]
```diff
@@ -0,0 +1,173 @@
+/* Copyright (c) 2018 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+/* See the file LICENSE for further information */
+#![no_std]
+
```
Comment (rjoleary) [inline src/soc/sifive/fu540/src/ddrregs.rs]: ddrregs and phy are two separate modules, so I think you have to import phy here with something like:

import create::phy::*;

File: src/soc/sifive/fu540/src/phy.rs [hunk 2]
```diff
@@ -0,0 +1,1220 @@
+/*:u32 =Copyright (c) 2018 SiFive, Inc */
+/*:u32 =SPDX-License-Identifier: Apache-2.0 */
+/*:u32 =SPDX-License-Identifier: GPL-2.0-or-later */
+/*:u32 =See the file LICENSE for further information */
+
+const DENALI_PHY_00_DATA:u32 =0x31706542;
```
Comment (rjoleary) [inline src/soc/sifive/fu540/src/phy.rs]: And make all these pub.

File: src/soc/sifive/fu540/src/phy.rs [hunk 3]
```diff
@@ -0,0 +1,1220 @@
+/*:u32 =Copyright (c) 2018 SiFive, Inc */
+/*:u32 =SPDX-License-Identifier: Apache-2.0 */
+/*:u32 =SPDX-License-Identifier: GPL-2.0-or-later */
+/*:u32 =See the file LICENSE for further information */
```
Comment (rjoleary) [inline src/soc/sifive/fu540/src/phy.rs]: ":u32 =" -- I think you went a bit overboard with find-and-replace in here.

Comment (rminnich) [reply src/soc/sifive/fu540/src/phy.rs]: ha ha I man I was tired when I did that :-)

**Pull request merged**